journald/journald.cpp

```diff
@@ -176,6 +287,16 @@ int main(int argc, char** argv)
     LOG(WARNING) << warning.message;
   }
 
+  // If the `--user` flag is set, change the UID of this process to that user.
+  if (flags.user.isSome()) {
+    Try<Nothing> result = os::su(flags.user.get());
+
+    if (result.isError()) {
+      EXIT(EXIT_FAILURE)
+        << ErrnoError("Failed to switch user for journald logger").message;
```

Comment:
This is more important for the `logrotate` portion, as logging will break if you don't switch the user appropriately.

I don't think journald has any restrictions on what user uses their APIs.